kroxylicious-operator/src/test/java/io/kroxylicious/operator/ProxyServiceTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.operator;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+import io.fabric8.kubernetes.api.model.Service;
+
+import io.kroxylicious.crapi.v1alpha1.KafkaProxy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ProxyServiceTest {
```

Comment:
I think we want unit tests for the dependent resource classes which covers the configuration possibilities (i.e. we try to cover a lot of the `KafkaProxy` parameter space and check that each dependent resource is what we expect).